src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/DeprecationCapability.cs

```diff
@@ -0,0 +1,82 @@
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class DeprecationCapability : IDeprecated
+    {
+        private static readonly string StarAll = VersionRangeFormatter.Instance.Format("p", VersionRange.Parse("*"), VersionRangeFormatter.Instance);
+        private static readonly string StarAllFloating = VersionRangeFormatter.Instance.Format("p", VersionRange.Parse("*-*"), VersionRangeFormatter.Instance);
+
+        public DeprecationCapability(PackageDeprecationMetadataContextInfo deprecatedInfo)
+        {
+            _deprecated = deprecatedInfo;
+        }
+
+        private PackageDeprecationMetadataContextInfo _deprecated;
+        public PackageDeprecationMetadataContextInfo DeprecationMetadata
+        {
+            get => _deprecated;
+            private set => _deprecated = value;
+        }
+
+        public bool IsDeprecated => DeprecationMetadata != null;
+        public string? AlternatePackageText
+        {
+            get
+            {
+                if (DeprecationMetadata.AlternatePackage == null)
+                {
+                    return null;
+                }
+
+                // pretty print
+                string versionString = VersionRangeFormatter.Instance.Format("p", DeprecationMetadata.AlternatePackage.VersionRange, VersionRangeFormatter.Instance);
+
+                if (StarAll.Equals(versionString, StringComparison.InvariantCultureIgnoreCase) || StarAllFloating.Equals(versionString, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return DeprecationMetadata.AlternatePackage.PackageId;
+                }
+
+                return $"{DeprecationMetadata.AlternatePackage.PackageId} {versionString}";
+            }
+        }
+
+        public string PackageDeprecationReasons
+        {
+            get
+            {
+                if (DeprecationMetadata.Reasons == null || !DeprecationMetadata.Reasons.Any())
+                {
+                    return Resources.Label_DeprecationReasons_Unknown;
+                }
+                else if (DeprecationMetadata.Reasons.Contains("CriticalBugs", StringComparer.OrdinalIgnoreCase))
+                {
+                    if (DeprecationMetadata.Reasons.Contains("Legacy", StringComparer.OrdinalIgnoreCase))
+                    {
+                        return Resources.Label_DeprecationReasons_LegacyAndCriticalBugs;
+                    }
+                    else
+                    {
+                        return Resources.Label_DeprecationReasons_CriticalBugs;
+                    }
+                }
+                else if (DeprecationMetadata.Reasons.Contains("Legacy", StringComparer.OrdinalIgnoreCase))
+                {
+                    return Resources.Label_DeprecationReasons_Legacy;
+                }
+                else
+                {
+                    return Resources.Label_DeprecationReasons_Unknown;
+                }
```

Comment:
Consider looping through Reasons once and tracking what we want to know and short circuiting if we find it. Currently, by doing a Reasons.Contains, we must iterate the list for each of the questions and iterate to the end even if we already know our answer.

Example in quick pseudocode:

bool isCritical, isLegacy;
For each reason in Reasons:
  Is reason Critical?
    then isCritical = true
  Is reason Legacy?
    then isLegacy = true
  
  Are isCritical and isLegacy both true? If so, exit loop.
End For Loop

if isCritical and isLegacy?
  then return CriticalLegacyCombo
else if isCritical?
  then return Critical
else if isLegacy?
  then return Legacy
else
  // default case
  then return Unknown